tests/default.nf.test

```diff
@@ -0,0 +1,274 @@
+                "FCS1" : [
+                    "FILE": "test-only.taxa.tsv",
+                    "OUT" : "all.taxa.tsv"
+                ],
+                "FCS2" : [
+                    "FILE": "test-only.gxi",
+                    "OUT": "all.gxi"
+                ],
+                "FCS3" : [
+                    "FILE": "test-only.gxs",
+                    "OUT": "all.gxs"
+                ],
+                "FCS4" : [
+                    "FILE": "test-only.meta.jsonl",
+                    "OUT": "all.meta.jsonl"
+                ],
+                "FCS5" : [
+                    "FILE": "test-only.blast_div.tsv.gz",
+                    "OUT": "all.blast_div.tsv.gz"
+                ]
+            ]
+
+            new File("${projectDir}/FCS_gx").mkdir()
+
+            for (String dbKey : fcs_database.keySet()) {
+                def db = fcs_database.get(dbKey)
+
+                println "\nDownloading the FCS file: ${db['FILE']}..."
+                def command2 = ['bash', '-c', "wget -cq https://ftp.ncbi.nlm.nih.gov/genomes/TOOLS/FCS/database/test-only/${db['FILE']} -O ${projectDir}/FCS_gx/${db['OUT']}"]
+                def process2 = command2.execute()
+                process2.waitFor()
+
+                if (process2.exitValue() != 0) {
+                    throw new RuntimeException("Error - failed to download ${dbKey}: ${process2.err.text}")
+                }
+
+                println "Done"
+            }
+
+            // Databases: The wierd ones
+
+            new File("${projectDir}/NCBI_TAXONOMY").mkdir()
+            new File("${projectDir}/DIAMOND").mkdir()
+            def other_dbs = [
+                "NCBI_TAXONOMY_DB": [
+                    "COMMAND":
+                        "curl -L --retry 5 --retry-delay 10 https://ftp.ncbi.nih.gov/pub/taxonomy/new_taxdump/new_taxdump.tar.gz --output new_taxdump.tar.gz && tar -C ${projectDir}/NCBI_TAXONOMY -xzf new_taxdump.tar.gz"
+                ],
+                "ASCC_DIAMOND_DB": [
+                    "COMMAND":
+                        "curl https://dp24.cog.sanger.ac.uk/ascc/diamond.dmnd -o ${projectDir}/DIAMOND/diamond.dmnd"
+                ]
+            ]
+
+            for (String dbKey : other_dbs.keySet()) {
+                def db = other_dbs.get(dbKey)
+
+                println "\nDownloading the OTHER DB's: ${dbKey}..."
+                def command3 = ['bash', '-c', "${db['COMMAND']}"]
+                def process3 = command3.execute()
+                process3.waitFor()
+
+                if (process3.exitValue() != 0) {
+                    throw new RuntimeException("Error - failed to download ${dbKey}: ${process3.err.text}")
+                }
+
+                println "Done"
+            }
+
+            // Useful for debugging
+            // String result = "ls -lha ".execute().text
+            // println result.toUpperCase()
+
+        }
+
+        when {
+            params {
+                outdir = "$outputDir"
```

Comment:
Set downloaded database param paths here - point to their downloaded locations in `launchDir`

Reply:
Done